lib/datadog/tracing/contrib/opensearch/patcher.rb

```diff
@@ -0,0 +1,120 @@
+# frozen_string_literal: true
+
+require_relative '../../metadata/ext'
+require_relative '../analytics'
+require_relative 'ext'
+require_relative '../ext'
+require_relative '../integration'
+require_relative '../patcher'
+
+module Datadog
+  module Tracing
+    module Contrib
+      module OpenSearch
+        # Patcher enables patching of 'opensearch' module.
+        module Patcher
+          include Contrib::Patcher
+
+          module_function
+
+          def patch
+            require 'uri'
+            require 'json'
+            require_relative 'quantize'
+
+            ::OpenSearch::Transport::Client.prepend(Client)
+          end
+
+          # Patches OpenSearch::Transport::Client module
+          module Client
+            # rubocop:disable Metrics/MethodLength
+            def perform_request(*args)
+              service = datadog_configuration[:service_name]
+              method = args[0]
+              path = args[1]
+              params = args[2]
+              body = args[3]
```

Comment:
`body` and `params` get overridden in the `tag_params` and `tag_body` methods, so it didn't work to remove all of the parameters back into the `def perform_request` line. I could move them all back to the line and then reset `body` and `params` separately?

Reply:
Reading the code in `tag_params`:

```
params = JSON.generate(params) unless params.is_a?(String)
```

This shouldn't mutate the original params. The arg `params` in this function is a local variable, that points to the params passed in (what we don't want to alter.) However, `JSON.generate` returns a new object, and doing `params=` changes the variable to point at that new object; it does not modify the old `params` object it used to point at. It should be okay.

It would be concerning if we did something like `params.reject!` where a `!` signifies a modification to the object being passed in. There are other functions that can also mutate an object (even without a `!`). It's good that you're being considerate that this is not happening here.

Reply:
Ahh I see, thanks!